### PR TITLE
add sql console section

### PR DESCRIPTION
**Pull request by cfahlgren1**
https://moon-ci-docs.huggingface.co/docs/hub/pr_1434/en/datasets-viewer-sql-console

image

### CODE REVIEW

File: docs/hub/_toctree.yml [hunk 1]
```diff
@@ -208,6 +208,8 @@
         title: Configure the Dataset Viewer
       - local: datasets-viewer-embed
         title: Embed the Dataset Viewer in a webpage
+      - local: datasets-viewer-sql-console
+        title: "SQL Console: Query datasets in the browser"
```
Comment (lhoestq) [inline docs/hub/_toctree.yml:212]: this should be enough

```suggestion
        title: "SQL Console"
```

File: docs/hub/datasets-viewer-sql-console.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# SQL Console: Query datasets in the browser
```
Comment (lhoestq) [inline docs/hub/datasets-viewer-sql-console.md:1]:
```suggestion
# SQL Console
```

File: docs/hub/datasets-viewer-sql-console.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+</p>
+
+
+Through the SQL Console, you can:
+
+- Run [DuckDB SQL queries](https://duckdb.org/docs/sql/query_syntax/select) on the dataset (_checkout [SQL Snippets](https://huggingface.co/spaces/cfahlgren1/sql-snippets) for useful queries_) 
+- Share results of the query with others via a link (_check out [this example](https://huggingface.co/datasets/gretelai/synthetic-gsm8k-reflection-405b?sql_console=true&sql=FROM+histogram%28%0A++train%2C%0A++topic%2C%0A++bin_count+%3A%3D+10%0A%29)_) 
+- Download the results of the query to a parquet file 
+- Embed the results of the query in your own webpage using an iframe 
+
```
Comment (lhoestq) [inline docs/hub/datasets-viewer-sql-console.md:21]: Maybe add a section with examples ? like a code snippet + like per example

some ideas: 

1. filter data (by value, and/or with string matching)
1. group by + count labels
1. concatenate and format strings from multiple columns
1. extract strings with regex

This is also the kind of content we can reuse for the duckdb cli/python client docs on HF at https://huggingface.co/docs/hub/main/datasets-duckdb

**Review by lhoestq — COMMENTED** [review]
Nice ! It's still a bit empty, I added a comment of content that could be useful (happy to help on this if needed)

File: docs/hub/datasets-viewer-sql-console.md (hunk 3 above)
Comment (cfahlgren1) [reply docs/hub/datasets-viewer-sql-console.md:21]: good point yeah I left it smaller since it was part of the original dataset viewer docs, but not that it has it's own page probably can use the content and examples 👍

File: docs/hub/datasets-viewer.md [hunk 4]
```diff
@@ -20,6 +20,11 @@ Similarly, if you select one class from a categorical column, it will show only
 
 You can search for a word in the dataset by typing it in the search bar at the top of the table. The search is case-insensitive and will match any row containing the word. The text is searched in the columns of `string`, even if the values are nested in a dictionary or a list.
 
+## Run SQL queries on the dataset
+
+You can run SQL queries on the dataset in the browser using the SQL Console.
```
Comment (julien-c) [inline docs/hub/datasets-viewer.md:25]:
```suggestion
You can run SQL queries on the dataset in the browser using the SQL Console. This feature also leverages our [auto-conversion to Parquet](datasets-viewer#access-the-parquet-files). 
```

File: docs/hub/datasets-viewer-sql-console.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+<Tip>
+You can also use the DuckDB CLI to query the dataset via the `hf://` protocol. See the <a href="https://huggingface.co/docs/hub/en/datasets-duckdb" target="_blank" rel="noopener noreferrer">DuckDB CLI documentation</a> for more information.
```
Comment (julien-c) [inline docs/hub/datasets-viewer-sql-console.md:23]:
```suggestion
You can also use a local DuckDB CLI to query the dataset via the `hf://` protocol. See the <a href="https://huggingface.co/docs/hub/en/datasets-duckdb" target="_blank" rel="noopener noreferrer">DuckDB CLI documentation</a> for more information.
```

you can already document the "Copy for DuckDB CLI" feature here IMO -- as it should be 🚢 soon

**Review by julien-c — COMMENTED** [review]
cool cool!

File: docs/hub/datasets-viewer-sql-console.md (hunk 5 above)
Comment (cfahlgren1) [reply docs/hub/datasets-viewer-sql-console.md:23]: good idea

File: docs/hub/datasets-viewer-sql-console.md (hunk 3 above)
Comment (cfahlgren1) [reply docs/hub/datasets-viewer-sql-console.md:21]: let me know what you think about the new examples

**Comment by cfahlgren1** [comment]
I put back the more verbose h1 so the seo title is better, but open to change

image

File: docs/hub/datasets-viewer-sql-console.md [hunk 6]
```diff
@@ -20,5 +20,111 @@ Through the SQL Console, you can:
 - Embed the results of the query in your own webpage using an iframe 
 
 <Tip>
-You can also use a local DuckDB CLI to query the dataset via the `hf://` protocol. See the <a href="https://huggingface.co/docs/hub/en/datasets-duckdb" target="_blank" rel="noopener noreferrer">DuckDB CLI documentation</a> for more information.
-</Tip>
\ No newline at end of file
+You can also use the DuckDB CLI to query the dataset via the `hf://` protocol. See the <a href="https://huggingface.co/docs/hub/en/datasets-duckdb" target="_blank" rel="noopener noreferrer">DuckDB CLI documentation</a> for more information. The SQL Console provides a convenient `Copy to DuckDB CLI` button that generates the SQL query for creating views and executing your query in the DuckDB CLI.
```
Comment (severo) [inline docs/hub/datasets-viewer-sql-console.md:23]: the DuckDB CLI-> a local DuckDB CLI again?

Comment (cfahlgren1) [reply docs/hub/datasets-viewer-sql-console.md:23]: Yes, probably could make this a little better 🤔

File: docs/hub/datasets-viewer-sql-console.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    <img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/sql_console/bar-struct-length-dark.png"/>
+</div>
+
+In the query, we can use the `len` function to get the length of the `reasoning_chains` column and the `bar` function to create a bar chart of the reasoning lengths.
+
+```sql
+select len(reasoning_chains) as reason_len, bar(reason_len, 0, 100), * 
+from train
+where reason_len > 10
+order by reason_len desc
```
Comment (lhoestq) [inline docs/hub/datasets-viewer-sql-console.md:87]:
```suggestion
SELECT len(reasoning_chains) AS reason_len, bar(reason_len, 0, 100), * 
FROM train
WHERE reason_len > 10
ORDER BY reason_len DESC
```

File: docs/hub/datasets-viewer-sql-console.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+<div class="flex justify-center">
+    <img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/sql_console/histogram-simple.png"/>
+    <img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/sql_console/histogram-simple-dark.png"/>
+</div>
+<p class="text-sm text-center italic">
+    Learn more about the `histogram` function and parameters <a href="https://cfahlgren1-sql-snippets.hf.space/histogram" target="_blank" rel="noopener noreferrer">here</a>.
+</p>
+
+```sql
+from histogram(train, len(reasoning_chains))
```
Comment (lhoestq) [inline docs/hub/datasets-viewer-sql-console.md:107]:
```suggestion
FROM histogram(train, len(reasoning_chains))
```

File: docs/hub/datasets-viewer-sql-console.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+- Embed the results of the query in your own webpage using an iframe 
+
+<Tip>
+You can also use the DuckDB CLI to query the dataset via the `hf://` protocol. See the <a href="https://huggingface.co/docs/hub/en/datasets-duckdb" target="_blank" rel="noopener noreferrer">DuckDB CLI documentation</a> for more information. The SQL Console provides a convenient `Copy to DuckDB CLI` button that generates the SQL query for creating views and executing your query in the DuckDB CLI.
+</Tip>
+
+
+## Examples
+
+### Leakage Detection
```
Comment (lhoestq) [inline docs/hub/datasets-viewer-sql-console.md:29]: nit, but I'd keep the example for the end since it's a more complex one

File: docs/hub/datasets-viewer-sql-console.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+    CASE 
+        WHEN total_count > 0 THEN (overlap_count * 100.0 / total_count)
+        ELSE 0
+    END AS overlap_percentage
+FROM overlapping_rows, total_unique_rows;
+```
+
+### Filtering
+
+The SQL Console makes filtering datasets really easily. For example, if you want to filter the `SkunkworksAI/reasoning-0.01` dataset for instructions and responses with a reasoning length of at least 10, you can use the following query:
```
Comment (lhoestq) [inline docs/hub/datasets-viewer-sql-console.md:74]:
```suggestion
The SQL Console makes filtering datasets really easy. For example, if you want to filter the `SkunkworksAI/reasoning-0.01` dataset for instructions and responses with a reasoning length of at least 10, you can use the following query:
```

File: docs/hub/datasets-viewer-sql-console.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+## Examples
+
+### Leakage Detection
+
+Leakage detection is the process of identifying whether data in a dataset is present in multiple splits, for example, whether the test set is present in the training set.
+
+<div class="flex justify-center">
+    <img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/sql_console/leakage-detection.png"/>
+    <img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/sql_console/leakage-detection-dark.png"/>
+</div>
```
Comment (lhoestq) [inline docs/hub/datasets-viewer-sql-console.md:36]: feel free to add links to the SQL console with the code from the example

**Review by lhoestq — APPROVED** [review]
added my final comments, otherwise lgtm :)

File: docs/hub/datasets-viewer-sql-console.md (hunk 9 above)
Comment (cfahlgren1) [reply docs/hub/datasets-viewer-sql-console.md]: done

File: docs/hub/datasets-viewer-sql-console.md (hunk 11 above)
Comment (cfahlgren1) [reply docs/hub/datasets-viewer-sql-console.md]: will update this when the upcoming pr is merged
